### PR TITLE
2017-10-18: Commerce frontend updates

### PR DESCRIPTION
**Pull request by zfox23**
A few changes here:
1. Add a `showDebugButtons` switch in the QML which dictates whether or not the Debug buttons show up on the "Help" tab of the Wallet
2. Fix "Recent Activity" indentation
3. Fixed login button JavaScript injection on an individual item page when not logged in
4. Small update to Help text for one answer
5. When clicking on "Show Backup Instructions" on the Security page, show the "Instructions Opened on Desktop" popup in Disabled Gray
6. Modify LowerKeyboard button location and image
7. Add "What does 'Rez' mean?" link in Checkout

# Test Plan

## Prerequisites

Currently, you must manually enable "commerce" by adding `"commerce": true,` in your Interface.json (with Interface not running), or by opening Interface Javascript console and evaluating `Settings.setValue("commerce", true)` and restarting.

If your wallet is already set up OR if you have previously set up a wallet on a different build (and thus you have a record of your wallet on the backend), you must perform the following steps to rest your wallet:
1. Open the Wallet app. If you're just resetting your wallet because there's a record of a previous wallet on the backend, go through setup.
2. Go to the "Help" tab in the Wallet app
3. Press the red "DBG: RST Wallet" button at the top of the window

If your wallet is set up but broken (e.g., from testing something that didn't work) such that you are prompted for a passphrase that is never accepted, you need to:
1. Quit interface and delete `%appdata%/Interface <build #>/<username.hifikey>`
2. Restart Interface, set up wallet, and then press the red `DBG: RST WALLET` button as above. Now you are in a truly clean state for wallet setup.

## Tests
Much of the changes here are tiny layout changes or invisible. These changes don't necessitate testing. For the others...

1. Go to your Sandbox
2. Set up your Wallet
3. Log out of High Fidelity
4. Open the Marketplace
5. Ensure you see a green "Log In" banner toward the top of the Marketplace page
6. Click on an individual item (the first one you see is fine) to navigate to the individual item's page
7. Ensure you see a green "Log In" banner toward the top of the page
8. Click the "Log In" button and log in
9. Search the Marketplace for "Test Beach Ball"
10. Buy the Test Beach ball and wait on the confirmation page
11. Ensure you see a "What does 'Rez' mean?" link below the red Rez button
12. Click the "What does 'Rez' mean?" link
13. Verify that you see the "Introduction to Purchases" tutorial. Go through the 2-step tutorial.
14. When you finish the tutorial, verify that you return to the Purchase Confirmation page.
15. Navigate to My Purchases
16. Verify that, upon navigating to My Purchases, you _do not_ see the "Introduction to Purchases" tutorial.
17. Don your HMD
18. Open your Wallet, then navigate to the Security tab. Tap to change your passphrase.
19. When you see the virtual keyboard, verify that you see a "Lower Keyboard" button at the top right of the keyboard. Verify that the button works to lower the keyboard.
20. Cancel this screen and navigate back to the Security tab of the Wallet
21. Click the "View Backup Instructions" link, and verify that the button changes to a disabled button with the text "Instructions Opened on Desktop" for a couple of seconds.

### CODE REVIEW

**Comment by Nex-Pro** [comment]
Awaiting build.

**Comment by zfox23** [comment]
Ah shoot, somehow there's a merge conflict - gotta fix that

**Comment by Nex-Pro** [comment]
Tick tock, come on Jenkins!

File: interface/resources/qml/hifi/commerce/wallet/Help.qml [hunk 1]
```diff
@@ -25,6 +25,7 @@ Item {
 
     id: root;
     property string keyFilePath;
+    property bool showDebugButtons: true;
```
Comment (howard-stearns) [inline interface/resources/qml/hifi/commerce/wallet/Help.qml]: Is the idea here is that we can edit the .qml file on the demo machine to remove the button?

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4276/HighFidelity-Beta-PR11631-aa4821593335da89bf3b388cad5cd1eeeedaff87.dmg

File: interface/resources/qml/hifi/commerce/wallet/Help.qml (hunk 1 above)
Comment (zfox23) [reply interface/resources/qml/hifi/commerce/wallet/Help.qml]: That is indeed the idea.

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/4276/HighFidelity-Beta-PR11631-aa4821593335da89bf3b388cad5cd1eeeedaff87.exe

**Comment by Nex-Pro** [comment]
T E S T I N G   Win10

**Comment by Nex-Pro** [comment]
Tested win10


### PASS



Go to your Sandbox

Set up your Wallet **DONE**

Log out of High Fidelity

Open the Marketplace

Ensure you see a green "Log In" banner toward the top of the Marketplace page **SUCCESS**

Click on an individual item (the first one you see is fine) to navigate to the individual item's page **DONE**

Ensure you see a green "Log In" banner toward the top of the page **SUCCESS**

Click the "Log In" button and log in

Search the Marketplace for "Test Beach Ball"

Buy the Test Beach ball and wait on the confirmation page

Ensure you see a "What does 'Rez' mean?" link below the red Rez button **SUCCESS**


Click the "What does 'Rez' mean?" link

Verify that you see the "Introduction to Purchases" tutorial. Go through the 2-step tutorial. **SUCCESS** Is it always true for all items that the first will disappear ? This will need to change at some point. 

When you finish the tutorial, verify that you return to the Purchase Confirmation page. **SUCCESS**

Navigate to My Purchases

Verify that, upon navigating to My Purchases, you do not see the "Introduction to Purchases" tutorial. **SUCCESS**

Don your HMD

Open your Wallet, then navigate to the Security tab. 

Tap to change your passphrase.

When you see the virtual keyboard, verify that you see a "Lower Keyboard" button at the top right of the keyboard. 
Verify that the button works to lower the keyboard. **SUCCESS**

Cancel this screen and navigate back to the Security tab of the Wallet

Click the "View Backup Instructions" link, 
and verify that the button changes to a disabled button with the text "Instructions Opened on Desktop" for a couple of seconds. **SUCCESS**

**Comment by zfox23** [comment]
yay!

File: interface/resources/qml/hifi/commerce/wallet/Help.qml (hunk 1 above)
Comment (jherico) [reply interface/resources/qml/hifi/commerce/wallet/Help.qml]: not if you're running the wallet QML on the desktop (as opposed to the tablet).  The desktop QML is now loaded out of the QRC so changes to the QML file in the resources file aren't visible.  You'd be better off editing the OffscreenQmlSurface to read an environment variable and export it to the QQmlEngine root context, then using that exported variable here.

**Pull request merged**